app/controllers/business_entities_controller.rb

```diff
@@ -0,0 +1,110 @@
+class BusinessEntitiesController < ApplicationController
+  def index
+    authorize :business_entity
+    list_jurisdictions
+  end
+
+  def new
+    redirect_to office_business_entities_path if valid_business_entity || !jurisdiction
+    @business_entity = BusinessEntity.new
+    authorize @business_entity
+  end
+
+  def create
+    @business_entity = business_entity_service.build(business_entity_params)
+    authorize @business_entity
+    if @business_entity.save
+      redirect_to office_business_entities_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    authorize business_entity
+  end
+
+  def update
+    bes = business_entity_service
+    new_be = bes.check_update(business_entity_params)
+    authorize new_be
+
+    if bes.persist_update!(new_be)
+      redirect_to office_business_entities_path
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  helper_method def office
+    @office ||= Office.find(params[:office_id])
+  end
+
+  helper_method def business_entity
+    @business_entity ||= BusinessEntity.find(params[:id])
+  end
+
+  def valid_business_entity
+    be = BusinessEntity.find_by(office_id: params[:office_id],
+                                jurisdiction_id: params[:jurisdiction_id],
+                                valid_to: nil)
+    flash[:alert] = t('error_messages.create_be_exists', jurisdiction: be.jurisdiction.name,
+                                                         office: be.office.name) if be
+    be
+  end
```

Comment:
As per the comment on #534 Ideally we'd get a method on `BusinessEntity`, which would return a valid one for given `office` and `jurisdiction`. Then this could be as simple as:

```ruby
def valid_business_entity
  BusinessEntity.valid_for(office_id: params[:office_id], jurisdiction_id: params[:jurisdiction_id]).tap do |be|
    flash[:alert] = t('error_messages.create_be_exists', jurisdiction: be.jurisdiction.name, office: be.office.name) if be
  end
end
```

I personally don't have a problem with an abbreviated block parameter name, but I'll leave it up to you. 

Maybe the flash alert could be abstracted to a private method, so then it would be 

```ruby
flash[:alert] = business_entity_exists_alert(be) if be
```